rfcs/20230807-Unify-primitive-and-graph-verbose/README.md

```diff
@@ -0,0 +1,139 @@
+  profiling work.
+
+To achieve this, `get_verbose()` and `verbose_t` need to be extended and
+`get_graph_verbose()` will be removed.
+
+### Unified Verbose Format
+
+To better unify the msg log with primitive log, this proposal is to align with
+the marker `onednn_verbose` and add a new column `primitive/graph` to indicate
+where the verbose came from. Take execution profiling as an example, we can see
+the first line came from primitive and the second came from graph API:
+
+```bash
+# primitive execution verbose
+onednn_verbose,primitive,exec,cpu,convolution,brgconv_1x1:avx512_core,forward_inference,src_f32::blocked:acdb::f0 wei_f32:a:blocked:Acdb64a::f0 bia_f32:a:blocked:a::f0 dst_f32::blocked:acdb::f0,attr-scratchpad:user attr-post-ops:eltwise_relu ,alg:convolution_direct,mb2_ic32oc64_ih112oh112kh1sh1dh0ph0_iw112ow112kw1sw1dw0pw0,1.15503
+
+# compiled partition execution verbose
+onednn_verbose,graph,exec,cpu,100006,convolution_post_ops,CONV_0;ELTWISE_1,data:NCX; filter:OIX;;,in0_f32:0:strided:undef:2x32x112x112:401408s12544s112s1 in1_f32:1:strided:constant:64x32x1x1:32s1s1s1 in2_f32:2:strided:constant:64:1 out0_f32:4:strided:undef:2x64x112x112:802816s12544s112s1,fpm:strict,dnnl_backend,5.44092
+```
+
+Also, like the primitive API, the graph API also supports some checking info and
+the msg log will be shown as:
+
+```bash
+# primitive check for invalid u8f32f32 matmul primitive
+onednn_verbose,primitive,create:check,matmul,invalid datatype for accumulation,src/common/matmul.cpp:188
+
+# graph check for invalid u8f32f32 matmul op
+onednn_verbose,graph,graph:check,add_op,MatMul,given data type for input0 is u8 v.s. expected {f16,bf16,f32},src/graph/interface/op_schema.cpp:250
+```
+
+To achieve this, graph related msg need to be added in `verbose_msg.hpp` and
+`VPROF/VCHECK/VINFO` need to be adjusted. Duplicated macros like
+`VPROFGRAPH/VCHECKGRAPH/VINFOGRAPH` can be removed. Also, minor changes may
+happen in `verbose_converter` side to only convert verbose from primitive API.
+
+### Unified Verbose Header
+
+Verbose header will be extended for showing infos from graph API. Registered
+backends and the graph verbose template can be printed.
+
+```bash
+onednn_verbose,info,oneDNN v3.2.0 (commit 01630c9b5795d34d077c228b9e7cd89a72d6b5e5)
+onednn_verbose,info,cpu,runtime:OpenMP,nthr:128
+onednn_verbose,info,cpu,isa:Intel AVX-512 with Intel DL Boost
+onednn_verbose,info,gpu,runtime:none
+onednn_verbose,graph,info,backend,0:dnnl_backend
+onednn_verbose,primitive,info,template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
+onednn_verbose,graph,info,template:operation,engine,partition_id,partition_kind,op_names,data_formats,logical_tensors,fpmath_mode,backend,exec_time
+```
+
+## API and implementation
+
+The changes will only impact the implementation and keep both primitive and
+graph APIs unchanged.
```

Comment:
We did not expose a programming API to control `ONEDNN_GRAPH_VERBOSE`. But now to need to think about `set_verbose()` API defined at https://github.com/oneapi-src/oneDNN/blob/master/include/oneapi/dnnl/dnnl.hpp#L13057.

Reply:
good catch. 0,1,2 set through this API should be aligned with the environment variable.